### PR TITLE
add support for using ARCH constant in easyconfig files

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: test/framework/options.py [hunk 1]
```diff
@@ -492,12 +492,16 @@ def run_test(fmt=None):
...
                 pattern_lines = [
+                    r'^``ARCH``\s*``(aarch64|ppc64le|x86_64)``\s*CPU architecture .*',
+                    r'^``EXTERNAL_MODULE``.*',
                     r'^``HOME``.*',
                     r'``OS_NAME``.*',
                     r'``OS_PKG_IBVERBS_DEV``.*',
                 ]
             else:
                 pattern_lines = [
+                    r'^\s*ARCH: (aarch64|ppc64le|x86_64) \(CPU architecture .*',
```
Comment (Micket) [inline test/framework/options.py:503]: I'll admit I don't understand where these patterns occur, but it does look like an unmatched `\)` here?

Comment (bartoldeman) [reply test/framework/options.py:503]: I think missing the `\)` is probably sufficient to match but it would look nicer and tighter with the `\)`

Comment (boegel) [reply test/framework/options.py:503]: The `\)` (an escaped `)` is not required thanks to the `.*`, but I can use `.*\)` to avoid confusion, sure...

**Review by bartoldeman — APPROVED** [review]
LGTM

**Pull request merged**